integrations/terraform-cloud/main.py

```diff
@@ -125,31 +116,33 @@ async def resync_workspaces(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
 @ocean.on_resync(ObjectKind.RUN)
 async def resync_runs(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
     terraform_client = init_terraform_client()
+    WORKSPACE_CONCURRENCY = 15
 
     async def fetch_runs_for_workspace(
         workspace: dict[str, Any]
-    ) -> List[List[Dict[str, Any]]]:
-        return [
-            run
-            async for run in terraform_client.get_paginated_runs_for_workspace(
+    ) -> list[dict[str, Any]]:
+        try:
+            all_runs = []
+            async for runs in terraform_client.get_paginated_runs_for_workspace(
                 workspace["id"]
+            ):
+                all_runs.extend(runs)
+            return all_runs
+        except Exception as e:
+            logger.error(
+                f"Error fetching runs for workspace {workspace['id']}: {str(e)}"
             )
-        ]
+            return []
 
-    async def fetch_runs_for_all_workspaces() -> ASYNC_GENERATOR_RESYNC_TYPE:
-        async for workspaces in terraform_client.get_paginated_workspaces():
-            logger.info(
-                f"Received {len(workspaces)} batch workspaces... fetching its associated {kind}"
-            )
+    async for workspaces in terraform_client.get_paginated_workspaces():
+        logger.info(f"Getting {kind}s for {len(workspaces)} workspaces")
 
-            tasks = [fetch_runs_for_workspace(workspace) for workspace in workspaces]
-            for completed_task in asyncio.as_completed(tasks):
-                workspace_runs = await completed_task
-                for runs in workspace_runs:
-                    yield runs
+        runs_per_workspace = await process_in_queue(
+            workspaces, fetch_runs_for_workspace, concurrency=WORKSPACE_CONCURRENCY
+        )
```

Comment:
why in queue?

Reply:
to control what we process concurrently. Leaving it open was causing timeouts even with rate limiter